hammer/library/aws/elasticsearch.py

```diff
@@ -0,0 +1,356 @@
+
+
+class ESDomainChecker:
+    """
+        Basic class for checking Elasticsearch unencrypted and logging issues in account/region.
+        Encapsulates discovered Elasticsearch domains.
+        """
+
+    def __init__(self, account):
+        """
+        :param account: `Account` instance with Elasticsearch domains to check
+        """
+        self.account = account
+        self.domains = []
+
+    def get_domain(self, id):
+        """
+        :return: `Elasticsearch Domain` by id
+        """
+        for domain in self.domains:
+            if domain.name == id:
+                return domain
+        return None
+
+    def check(self, ids=None):
+        """
+        Walk through Elasticsearch domains in the account/region and put them to `self.domains`.
+
+        :param ids: list with Elasticsearch domain ids to check, if it is not supplied - all Elasticsearch domains must be checked
+
+        :return: boolean. True - if check was successful,
+                          False - otherwise
+        """
+        domain_details = []
+        try:
+            es_client = self.account.client("es")
+            if ids is None:
+                ids = []
+                domain_names_list = es_client.list_domain_names()["DomainNames"]
+                for domain_name in domain_names_list:
+                    ids.append(domain_name["DomainName"])
+
+            if ids is not None:
+                domain_details = es_client.describe_elasticsearch_domains(DomainNames=ids)["DomainStatusList"]
+
+        except ClientError as err:
+            if err.response['Error']['Code'] in ["AccessDenied", "UnauthorizedOperation"]:
+                logging.error(f"Access denied in {self.account} "
+                              f"(ec2:{err.operation_name})")
+            else:
+                logging.exception(f"Failed to describe elasticsearch domains in {self.account}")
+            return False
+
+        for domain_detail in domain_details:
+            is_logging = False
+            domain_encrypted = False
+            domain_name = domain_detail["DomainName"]
+            domain_id = domain_detail["DomainId"]
+            domain_arn = domain_detail["ARN"]
+            encryption_at_rest = domain_detail.get("EncryptionAtRestOptions")
+            node_to_node_encryption = domain_detail.get("NodeToNodeEncryptionOptions")
+            if encryption_at_rest and encryption_at_rest["Enabled"]:
+                domain_encrypted = True
+            elif node_to_node_encryption and node_to_node_encryption["Enabled"]:
+                domain_encrypted = True
+
+            logging_details = domain_detail.get("LogPublishingOptions")
+
+            if logging_details:
+                index_logs = logging_details.get("INDEX_SLOW_LOGS")
+                search_logs = logging_details.get("SEARCH_SLOW_LOGS")
+                error_logs = logging_details.get("ES_APPLICATION_LOGS")
+                if (index_logs and index_logs["Enabled"]) \
```

Comment:
Should we enable only one or all three? Let's discuss.

Reply:
Atleast one logging, so added for Error logs.